webui/react/src/hooks/useFeature.ts

```diff
@@ -20,19 +37,37 @@ interface FeatureHook {
 
 const useFeature = (): FeatureHook => {
   const info = useObservable(determinedStore.info);
-  return { isOn: (ValidFeature) => IsOn(ValidFeature, info) };
+  const featureSettings = useObservable(
+    userSettings
+      .get(featureSettingsConfig, FEATURE_SETTINGS_PATH)
+      .select((loadable) => Loadable.getOrElse(null, loadable)),
+  );
+  return { isOn: (feature: ValidFeature) => IsOn(feature, info, featureSettings) };
 };
 
 // Priority: Default state < config settings < user settings < url
-const IsOn = (feature: ValidFeature, info: DeterminedInfo): boolean => {
+const IsOn = (
+  feature: ValidFeature,
+  info: DeterminedInfo,
+  settings: FeatureSettingsConfig | null,
+): boolean => {
   const { featureSwitches } = info;
   // Read from default state
-  let isOn = FeatureDefault[feature];
+  let isOn: boolean = FeatureDefault[feature];
 
   // Read from config settings
   featureSwitches.includes(feature) && (isOn = true);
   featureSwitches.includes(`-${feature}`) && (isOn = false);
 
+  // Read from user settings
+  if (settings && feature in settings) {
+    const userValue = settings[feature];
+    // checks are split up bc typescript doesn't combine the typeguards properly
```

Comment:
quick typeescript playground link explaining what's happening in the code:
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wGIFECCAVAVQCUsBlOAXjgG8AoORACwENYAuOAcgDlgB3OAGEWsVJwA09OKDAAbAJboA+gDcATBx784WcMCjyAtsGTwAMopgSpUMEoBG85ABMnAc01FgqCAFcoCMBwAAoQELJwAEJOrshu1gC+cMyoiCjoANy0MACeYEEAaswKzhjAzDD+QVQA1sA5EABmcLn5TXDY+MRktLQA9H0tjIpwIzCMQVDevrIw7i15QY1QEIajEAC0MKkbG6Pwhsx1qb4uwI1OwM7S4AoI8jDicKeN0JXIFcCyOdmLHeWVKakYAwOZxVCCFAXNyUGhSBgIEQwAD8HHsYVk5WQcAAPnBkDMInjTs5zpdnFkGAwZAplOpUXB0eEsbj8YTWSSycgrpSqbYHDF3AymZjmNi8QTZETnmcLtyKbQEr0ZG80mh4DBvDAMKcELCABR1HIcIolMoVKpPVAgsFuVAcSXS82A4DA0HuCFQ+RuACUlAAfHCGJj4CK0RiWVRGsVrb0GFJ5M19db3eC4AAydNwI2jbEp22pTNwfMegDaRoAunAAIQUKiOjNZkvg8v1Ku1qicuVXP10KlwAZwKbODitYBuXysZypFDFm1lytwZwoTjwMApVLjFa+NyMeGMjGw5t21s5CtSJUMRNwZPztNFnNOOepu29-dIdVwFTFXzVZ8F08K15K8k2-WRfxrOs2SlRsvx-IIOxlUlu2cN9+wYEVYTA3990vOAlRsEF-GxEVFSAA

Reply:
Yeah, in order for it to work as one type Typescript would have to narrow the type of `settings` based on the latter checks which is very tricky.